File: app/src/main/java/com/lucciola/haskellanywhereforandroid/controller/UpDownButtonController.java
package com.lucciola.haskellanywhereforandroid.controller;

import com.lucciola.haskellanywhereforandroid.View;
import com.lucciola.haskellanywhereforandroid.ProgramManager;
import com.lucciola.haskellanywhereforandroid.R;

/**
 * Created by hiden on 2017/06/23.
 */

public class UpDownButtonController extends Controller {

    public UpDownButtonController(View arg0, ProgramManager arg2) {
        super(arg0, arg2);
    }

    @Override
    public void onClick(android.view.View v) {
        switch (v.getId()) {
            case R.id.up:
                this.view.setInputText(this.manager.upInputList());
                this.view.setInputCursorLast();
                break;
            case R.id.down:
                this.view.setInputText(this.manager.downInputList());
                this.view.setInputCursorLast();
                break;
        }
    }

}

File: app/src/main/java/com/lucciola/haskellanywhereforandroidkt/controller/UpDownButtonController.kt
package com.lucciola.haskellanywhereforandroidkt.controller

import com.lucciola.haskellanywhereforandroidkt.View
import com.lucciola.haskellanywhereforandroidkt.ProgramManager
import com.lucciola.haskellanywhereforandroidkt.R

/**
 * Created by hiden on 2017/06/23.
 */

class UpDownButtonController(arg0: View, arg2: ProgramManager) : Controller(arg0, arg2) {

    override fun onClick(v: android.view.View) {
        when (v.id) {
            R.id.up -> {
                this.view.inputText = this.manager.upInputList()
                this.view.setInputCursorLast()
            }
            R.id.down -> {
                this.view.inputText = this.manager.downInputList()
                this.view.setInputCursorLast()
            }
        }
    }

}
